### DISCUSSION TITLE
AR difference between library versions

### DISCUSSION THREAD

**Discussion by divclass1** [google/model-viewer#4924] [Q&A] [answered]
Hi,

I checked my 3D model in AR, but I see there is a difference with modelviewer libraries. As you can see the AR version looks not good on version 2.1.0

Now, when I upload the 3D model in the editor, I do see some warnings.. I have 2 questions:

1: How important are these warnings? Should I always make sure there are no warnings?

2: Is it possible that this 3D model was good with no warnings a couple months ago, but with new updates of the editor / libraries it's showing warnings?  is it possible that models that are good now, will show warnings in the future?

Thanks in advance :D

![new](https://github.com/user-attachments/assets/b701e0a0-d8f3-4f79-ab97-eb2806c6c6ba)
![old](https://github.com/user-attachments/assets/fdc2f3fb-07ca-4046-9939-dff3b97aafc7)

**Comment by elalish** [comment, marked as answer]
If you're talking about AR, it's really important to let us know which mode / what device you're using. I'm guessing this is Android WebXR (default), in which case this is the same look as in 3D. Don't use old versions of MV unless you have to. We fix a lot of bugs over time, so old versions are simply buggy, which is probably what you're seeing. 

As for glTF validator warnings - there are four levels. Errors: must fix. Warnings: pretty serious, you probably need to fix. Info, Hints: no big deal. They *might* change over time as the validator improves, but it's unlikely. Note, that validator is completely independent of MV.

**Reply by divclass1** [reply]
This happened on iPhones. I updated to the latest and the issue was solved. 

Thanks for explaining the warnings. I will make sure all the warnings get fixed. I appreciate your reply and answers. I love Modelviewer!

**Reply by divclass1** [reply]
Hi @elalish , another quick question. As there is a difference between GLB and USDZ,  is there also an online viewer like Modelviewer where I can upload my GLB file, and see how it will look in AR on iPhone? This would save time, now I have to see it in AR first, then make changes to the 3D model to tweak it, until it's perfect for iOS too. Would be nice if there is a viewer that immediately shows how it will look on iPhone, so you don't have to actually check it in AR.

Thanks!!

**Reply by elalish** [reply]
Indeed - we provide exactly that! Just drop your model into our [editor](https://modelviewer.dev/editor/) and click "Deploy Mobile" and scan the QR code with your iPhone.

**Reply by divclass1** [reply]
That works great, but my 3D designer doesn't have an iPhone. So it's always guessing for him how it will turn out on iPhone. That's the problem I'm facing😅

**Reply by elalish** [reply]
Alas, I don't know of any way around that. Part of our difficulty with USDZ is the only "spec" is what Apple has implemented, so it's difficult to validate with any other tool. I would just find the cheapest old used iPhone you can that runs the latest iOS.
